src/NHibernate/Impl/AbstractSessionImpl.cs

```diff
@@ -313,17 +313,26 @@ public ProcessHelper(AbstractSessionImpl session)
 			{
 				_session = session;
 				_context = new SessionIdLoggingContext(session.SessionId);
-				session.CheckAndUpdateSessionStatus();
-				_session._processing = true;
+				try
+				{
+					_session.CheckAndUpdateSessionStatus();
+					_session._processing = true;
+				}
+				catch
+				{
+					_context.Dispose();
+					_context = null;
+					throw;
+				}
 			}
 
 			public void Dispose()
 			{
+				_context.Dispose();
```

Comment:
If disposed is called multiple times, this will cause a null ref. We need a `?.` here.